core/src/main/java/org/apache/iceberg/IncrementalDataTableScan.java

```diff
@@ -61,10 +63,7 @@ public TableScan useSnapshot(long scanSnapshotId) {
 
   @Override
   public TableScan appendsBetween(long newFromSnapshotId, long newToSnapshotId) {
-    Preconditions.checkArgument(
-        table().snapshot(newFromSnapshotId) != null, "fromSnapshotId: %s does not exist", newFromSnapshotId);
-    Preconditions.checkArgument(
-        table().snapshot(newToSnapshotId) != null, "toSnapshotId: %s does not exist", newToSnapshotId);
+    validateSnapshotIdsRefinement(newFromSnapshotId, newToSnapshotId);
```

Comment:
refinement checks are only done in `IncrDataTableScan.appendsBetween` as it cannot be triggered through `DataTableScan.appendsBetween`